### PR TITLE
ci: update CI workflows and test script

### PR DESCRIPTION
**Pull request by wyardley**
# Description of the change
- Simplify find calls and remove extra greps
- Update `run_policy_tests.sh` to exit non-0 and print an error on failures (for local use)
- Update `actions/checkout` to `v4`
- Update setup-regal workflow to v1 (note: keeping lint version at `v0.11.0`, since the latest version seems to have a number of issues with the current config

## Checklist

Please make sure that the proposed change checks all the boxes below before requesting a review:

- [ ] Each new policy has corresponding tests.
- [x] All the tests pass.
- [x] The pull request is opened against the `main` branch.
- [x] The pull request is no longer marked as a draft.
- [x] You agree to license your contribution under the [MIT license](https://github.com/spacelift-io/spacelift-policies-example-library/blob/main/LICENSE) to Spacelift (not required for Spacelift employees).

If the proposed change is ready to be merged, please request a review from `@spacelift-io/solutions-engineering`. Someone will review and merge the pull request.

_Spacelift employees should request reviews from the relevant engineers and are allowed to merge pull requests after they got at least one approval._

Thank you for your contribution! 🙇

### CODE REVIEW

File: .github/workflows/continuous-integration.yaml [hunk 1]
```diff
@@ -12,9 +12,9 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
       - name: Setup Regal
-        uses: StyraInc/setup-regal@v0.2.0
+        uses: StyraInc/setup-regal@v1
         with:
           version: v0.11.0
```
Comment (wyardley) [inline .github/workflows/continuous-integration.yaml:19]: https://docs.styra.com/regal/rules/idiomatic/use-if
https://docs.styra.com/regal/rules/idiomatic/use-contains
https://docs.styra.com/regal/rules/idiomatic/boolean-assignment
https://docs.styra.com/regal/rules/imports/use-rego-v1

A lot of the lint warnings in newer regal are related to these findings - I could take a pass at updating some of those separately if there's interest.

**Comment by wyardley** [comment]
I don’t think it’s possible to request a review from @spacelift-io/solutions-engineering as a non-member, and mentions may also not work.

**Review by Yantrio — APPROVED** [review]

**Comment by Apollorion** [comment]
I like it, thanks for the contribution!

**Comment by Apollorion** [comment]
I like it, thanks for the contribution!

**Pull request merged**